### DISCUSSION TITLE
EntitiyFetcher for union

### DISCUSSION THREAD

**Discussion by GellertH** [Netflix/dgs-framework#1380] [Q&A]
Hello, is it possible to declare an entity fetcher on an interface object?
Sadly dgsData data fetching would not be enough since this is a federated setting. 
I would be able to resolve type and data by ID.

Example:
interface A @key(fields: "id") {
id: ID!
...
}
type B implements A {...}
type C implements A {...}

//this is a shared object across multiple services
type SomeObject {
  typeAField: A @provides(fields: "id")
}

**Comment by GellertH** [comment]
I tried declaring DgsEntityFetcher for the interface type as well as the implementing types, but they remain unused.

**Reply by archanasevak** [reply]
@GellertH did you figure this out, i am running into similar issue.
